wntr/tests/test_network.py

```diff
@@ -979,6 +979,14 @@ def test_json_pattern_dump(self):
         wn = wntr.network.WaterNetworkModel()
         wn.add_pattern('pat0', [0,1,0,1,0,1,0])
         self.wntr.network.write_json(wn, f'temp.json')
+        
+    def test_dict_with_leak(self):
+        # This covers a bug where writing controls to a dictionary broke if a leak was added
+        wn = wntr.network.WaterNetworkModel(self.inp_file)
+        junction_name = wn.junction_name_list[0]
+        junction = wn.get_node(junction_name)
+        junction.add_leak(wn, area=1, start_time=0, end_time=3600)
+        wn.to_dict()
```

Comment:
I suggest adding `from_dict` to make sure the leak can be recreated in a water network model.

Reply:
Good idea!